### PR TITLE
Update sliceviewer manual testing docs

### LINKED ISSUE #33228: Update sliceviewer manual testing docs
Found during manual testing of sliceviewer #33201

**Describe the outcome that is desired.**
Easier to follow step-by-step instructions with simulated or real data (not empty workspaces).

- Split single page up by workspace type
- Each page of instructions should be easy to follow from top to bottom without jumping around
- Use either links or include statements to share common sections
- MDHisto won't display number of bins as you can't rebin histo
- Add some simulated data to MDEvent 4D workspace to easily check transposition etc.
```
md_4D_1 = CreateMDWorkspace(Dimensions=4, Extents=[-1,1,-2,2,-3,3,-4,4], Names="H,K,L,E", Frames='HKL,HKL,HKL,General Frame',Units='r.l.u.,r.l.u.,r.l.u.,meV')
FakeMDEventData(InputWorkspace=md_4D_1, UniformParams='5e6') # 4D data
md_4D_2 = CreateMDWorkspace(Dimensions=4, Extents=[-0.5,0.5,-1,2,-3,2,-4,4], Names="H,K,L,E", Frames='HKL,HKL,HKL,General Frame',Units='r.l.u.,r.l.u.,r.l.u.,meV')
FakeMDEventData(InputWorkspace=md_4D_2, UniformParams='5e6') # 4D data
md_4D = md_4D_1 + md_4D_2

expt_info = CreateSampleWorkspace()
md_4D.addExperimentInfo(expt_info)
```
  - Add item to test peak overlays in 4D+
  - Section 8:
    - ROI: Check when ROI enabled that line plots button is enabled
  - Do delete workspace as last check
  - Do rename first then scale/multiply
  - Rename/delete peaks workspace when overlayed
  - Integrate peaks. Spherically symmetric peaks in non-orthogonal view show correctly:
    - create 2 workspaces of integegrated peaks one with sphere & one with ellipsoid
```
    ws = CreateMDWorkspace(Dimensions='3', Extents='-6,6,-4,4,-2,2',
                       Names='H,K,L', Units='r.l.u.,r.l.u.,r.l.u.',
                       Frames='HKL,HKL,HKL',
                       SplitInto='2', SplitThreshold='50')
expt_info = CreateSampleWorkspace()
ws.addExperimentInfo(expt_info)
SetUB(ws, 1,1,2,90,90,120)
# add peak
CreatePeaksWorkspace(InstrumentWorkspace='ws', NumberOfPeaks=0, OutputWorkspace='peaks')
CopySample(InputWorkspace='ws', OutputWorkspace='peaks', CopyName=False, CopyMaterial=False, CopyEnvironment=False, CopyShape=False)
p = mtd['peaks']
FakeMDEventData(ws, EllipsoidParams='1e+04,0,1,1,1,0,0,0,1,0,0,0,1,0.02,0.06,0.03,1', RandomSeed='3873875')
pk = p.createPeakHKL([0,1,1]) # axes aligned with viewing axes
p.addPeak(pk)
FakeMDEventData(ws, EllipsoidParams='1e+04,0,1,0,1,0,0,0,1,0,0,0,1,0.06,0.06,0.06,1', RandomSeed='3873875')
pk = p.createPeakHKL([0,1,0]) # axes aligned with viewing axes
p.addPeak(pk)

IntegratePeaksMD(InputWorkspace='ws', PeakRadius=0.6, BackgroundInnerRadius=0.6,
    BackgroundOuterRadius=1, PeaksWorkspace='peaks', OutputWorkspace='peaks_int_centroid_unfixed', 
    Ellipsoid=True, UseCentroid=False, FixMajorAxisLength = True, UseOnePercentBackgroundCorrection=False)
IntegratePeaksMD(InputWorkspace='ws', PeakRadius=0.6, BackgroundInnerRadius=0.6,
    BackgroundOuterRadius=1, PeaksWorkspace='peaks', OutputWorkspace='peaks_int_sphere', 
    Ellipsoid=False, UseOnePercentBackgroundCorrection=False)
```
- overlay both workspaces
- switch to non-ortho mode and both integration regions should match (closely)
- Checking non-axis aligned cuts produce data at expected positions (integer HKL):
```
ws = CreateMDWorkspace(Dimensions='3', Extents='-3,3,-3,3,-3,3',
                       Names='H,K,L', Units='r.l.u.,r.l.u.,r.l.u.',
                       Frames='HKL,HKL,HKL',
                       SplitInto='2', SplitThreshold='10')

# add fake Bragg peaks for primitive lattice in data
for h in range(-3,4):
    for k in range(-3,4):
        for l in range(-3,4):
            hkl = ",".join([str(x) for x in [h,k,l]])
            FakeMDEventData(ws, PeakParams='1e+02,' + hkl + ',0.02', RandomSeed='3873875')
 
BinMD(InputWorkspace=ws, AxisAligned=False,
    BasisVector0='[00L],U,0,0,1', 
    BasisVector1='[HH0],U,1,1,0',
    BasisVector2='[-HH0],U,-1,1,0',
    OutputExtents='-4,4,-4,4,-0.25,0.25', 
    OutputBins='101,101,1', OutputWorkspace='BinMD_out', NormalizeBasisVectors=False)
```

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
**Description of work.**

See issue #33228 for details - essentially restructure the docs so that the tests are more self-contained (or use links where tests should be repeated) and run sequentially.

Have removed the real SXD data (and picking of peaks etc.) as this added significantly to the time taken to test but not to the functionality tested.

**To test:**
The diff is not that helpful so better build the docs

1. Build dev-docs
2. Open sliceviewer.html
3. Perform manual testing - note already found issues #33394 and #33409 performing these tests.

Fixes #33228 

*This does not require release notes* because **is a dev doc change**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (DanielMurphy22) [inline dev-docs/source/Testing/SliceViewer/SliceViewer.rst:146]:
```suggestion
Test the :ref:`Nonorthogonal view<mantid:sliceviewer_nonortho>`
```

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (DanielMurphy22) [inline dev-docs/source/Testing/SliceViewer/SliceViewer.rst:175]:
```suggestion
Test the :ref:`Peak Overlay<mantid:sliceviewer_peaks_overlay>`
```

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (DanielMurphy22) [inline dev-docs/source/Testing/SliceViewer/SliceViewer.rst:137]:
```suggestion
4. Change the thickness of the integration width along the non-viewed axes.
```
The word thick brings you to the right part of the GUI.

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (DanielMurphy22) [inline dev-docs/source/Testing/SliceViewer/SliceViewer.rst:152]: I got this error on `md_3D`:

``` python
Traceback (most recent call last):
  File "/home/danielmurphy/mantid/qt/python/mantidqt/mantidqt/widgets/sliceviewer/view.py", line 425, in on_non_orthogonal_axes_toggle
    self.presenter.nonorthogonal_axes(state)
  File "/home/danielmurphy/mantid/qt/python/mantidqt/mantidqt/widgets/sliceviewer/presenter.py", line 365, in nonorthogonal_axes
    self.show_all_data_requested()
  File "/home/danielmurphy/mantid/qt/python/mantidqt/mantidqt/widgets/sliceviewer/presenter.py", line 239, in show_all_data_requested
    self.set_axes_limits(*limits)
  File "/home/danielmurphy/mantid/qt/python/mantidqt/mantidqt/widgets/sliceviewer/presenter.py", line 256, in set_axes_limits
    self.data_limits_changed()
  File "/home/danielmurphy/mantid/qt/python/mantidqt/mantidqt/widgets/sliceviewer/presenter.py", line 226, in data_limits_changed
    self.new_plot()  # automatically uses current display limits
  File "/home/danielmurphy/mantid/qt/python/mantidqt/mantidqt/widgets/sliceviewer/presenter.py", line 125, in new_plot_MDE
    dimension_indices=dimension_indices))
  File "/home/danielmurphy/mantid/qt/python/mantidqt/mantidqt/widgets/sliceviewer/view.py", line 296, in plot_MDH_nonorthogonal
    self.draw_plot()
  File "/home/danielmurphy/mantid/qt/python/mantidqt/mantidqt/widgets/sliceviewer/view.py", line 352, in draw_plot
    self.colorbar.update_clim()
  File "/home/danielmurphy/mantid/qt/python/mantidqt/mantidqt/widgets/colorbar/colorbar.py", line 264, in update_clim
    self._autoscale_clim()
  File "/home/danielmurphy/mantid/qt/python/mantidqt/mantidqt/widgets/colorbar/colorbar.py", line 298, in _autoscale_clim
    data = self.colorbar.mappable.get_array_clipped_to_bounds()
AttributeError: 'QuadMesh' object has no attribute 'get_array_clipped_to_bounds'
```

Comment (DanielMurphy22) [reply dev-docs/source/Testing/SliceViewer/SliceViewer.rst:152]: I get the same error if I change displayed axes while in Nonoortho view

Comment (DanielMurphy22) [reply dev-docs/source/Testing/SliceViewer/SliceViewer.rst:152]: I also get these QuadMesh errors with MDHistoworkspace

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (DanielMurphy22) [inline dev-docs/source/Testing/SliceViewer/SliceViewer.rst:207]:
```suggestion
1. Make a 3D MDHistoWorkspaces
```
The code block only makes a 3D

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (DanielMurphy22) [inline dev-docs/source/Testing/SliceViewer/SliceViewer.rst:223]: It didn't close, but dynamic rebinning seems fine... I think the rebinning seemed fine because it had produced an ``md_4d_svrebinned_sv_rebinned`` workspace. But it doesn't close, even if I open ``md_4D_svrebinned``, do not zoom and delete md_4D

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (DanielMurphy22) [inline dev-docs/source/Testing/SliceViewer/SliceViewer.rst:271]: testing these next few lines was blocked by the QuagMesh error!

**Review by DanielMurphy22 — CHANGES_REQUESTED** [review]
The new instructions are really good and I appreciate the effort you've put in to the new format! I've some nitpicks and a few problems. The only big thing is the QuadMesh error for non-orthogonal view. 

As a reminder to myself, after lunch I should continue with testing from `2. ADS observer for peak overlay`

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (RichardWaiteSTFC) [reply dev-docs/source/Testing/SliceViewer/SliceViewer.rst:223]: Ah yes that's related to a memory leak - we know about that but I can't find the issue...

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (RichardWaiteSTFC) [reply dev-docs/source/Testing/SliceViewer/SliceViewer.rst:27]: I think you might need to build all the docs as they are links to the sliceviewer interface docs

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (RichardWaiteSTFC) [reply dev-docs/source/Testing/SliceViewer/SliceViewer.rst:207]: Ah good point  - the 2D test is covered by the `md_4D_svrebinned` stuff

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (DanielMurphy22) [inline dev-docs/source/Testing/SliceViewer/SliceViewer.rst:341]: In sections 3. and 4. of the Specific tests you refer to ``md_3D`` as ``MD_3D``. This is mostly not a problem apart from the CloneWorkspace code line.

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (DanielMurphy22) [inline dev-docs/source/Testing/SliceViewer/SliceViewer.rst:423]:
```suggestion
2. Open `` ws_4D`` in sliceviewer. Confirm that when the Energy axis is viewed (as X or Y) the non-orthogonal view is disabled.
```
From the code snippet it's obvious but testers probably have lots in the ADS at this point!

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (DanielMurphy22) [inline dev-docs/source/Testing/SliceViewer/SliceViewer.rst:26]: Just for the very first time, can we say:
```suggestion
1. Load the workspace. Right-click on the workspace and select Show Slice Viewer.
```

**Review by DanielMurphy22 — CHANGES_REQUESTED** [review]
The specific section is really well written!

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (RichardWaiteSTFC) [reply dev-docs/source/Testing/SliceViewer/SliceViewer.rst:79]: I can't reproduce this - what OS are you on?

Comment (DanielMurphy22) [reply dev-docs/source/Testing/SliceViewer/SliceViewer.rst]: Ubuntu 18.04

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (RichardWaiteSTFC) [reply dev-docs/source/Testing/SliceViewer/SliceViewer.rst:221]: Have made a new issue for this #33434

File: docs/source/workbench/sliceviewer.rst
Comment (DanielMurphy22) [inline docs/source/workbench/sliceviewer.rst:71]:
```suggestion
.. _sliceviewer_peak_actions:

```

File: docs/source/workbench/sliceviewer.rst
Comment (DanielMurphy22) [inline docs/source/workbench/sliceviewer.rst:96]:
```suggestion

.. _sliceviewer_nonortho:

```

File: docs/source/workbench/sliceviewer.rst
Comment (DanielMurphy22) [inline docs/source/workbench/sliceviewer.rst:116]:
```suggestion
.. _sliceviewer_roi:

```

**Review by DanielMurphy22 — CHANGES_REQUESTED** [review]
Sorry, last thing! I can't test the links all work as they point to the webpages on the docs in the sections you've added here. Fingers crossed they'll work when the online docs and dev-docs are rebuilt, but thought adding these newlines would give us the best chance! Promise I'll approve after!

**Review by DanielMurphy22 — APPROVED** [review]
Super good job! The Sliceviewer testing docs are great now!

File: dev-docs/source/Testing/SliceViewer/SliceViewer.rst
Comment (RichardWaiteSTFC) [reply dev-docs/source/Testing/SliceViewer/SliceViewer.rst:39]: I've made a note on the issue that the extent of the cut along non-spectrum axis e.g. TOF is incorrect - it seems to grab more than the axes limits.

**Review by DavidFair — APPROVED** [review]

**Comment by DavidFair** [comment]
A nice improvement on the testing docs :+1:

**Pull request merged**
